packages/media-utils/src/components/media-upload/index.js

```diff
@@ -400,6 +400,8 @@ class MediaUpload extends Component {
 		if ( onClose ) {
 			onClose();
 		}
+
+		this.frame.detach();
```

Comment:
This should work, but it looks like the `detach()` method calls close as well, so you might be able to bind the 'close' event directly to `detach` similar to how the `EditAttachments` frame works: https://github.com/WordPress/wordpress-develop/blob/trunk/src/js/media/views/frame/edit-attachments.js#L68

Reply:
What would be the difference between binding `detach` directly to the event vs calling it from the binded callback?

I think they will be the same in this code base.